diff -upNr compiz-0.8.2.orign/configure.ac compiz-0.8.2/configure.ac
--- compiz-0.8.2.orign/configure.ac	2009-03-01 13:10:46.000000000 +0100
+++ compiz-0.8.2/configure.ac	2009-08-22 17:25:51.905352173 +0200
@@ -331,7 +331,7 @@ if test "x$use_gtk" = "xyes"; then
 
     if test "x$use_gnome" = "xyes"; then
       PKG_CHECK_MODULES(GNOME_WINDOW_SETTINGS,
-			gnome-window-settings-2.0 gnome-desktop-2.0,
+			gnome-window-settings-2.0 gnome-desktop-2.0 gconf-2.0,
 			[use_gnome=yes], [use_gnome=no])
     fi
 
