diff -up compiz-0.8.2/src/window.c.stacking-order compiz-0.8.2/src/window.c
--- compiz-0.8.2/src/window.c.stacking-order	2010-06-22 19:07:59.512240198 -0400
+++ compiz-0.8.2/src/window.c	2010-06-22 19:08:57.905237401 -0400
@@ -4290,14 +4290,19 @@ updateWindowAttributes (CompWindow      
 		if (p->id == w->screen->display->activeWindow)
 		    break;
 
-	    /* window is above active window so we should lower it */
-	    if (p)
+	    /* window is above active window so we should lower it, assuming
+	     * that is allowed (if, for example, our window has the "above" state,
+	     * then lowering beneath the active window may not be allowed.)
+	     */
+	    if (p && validSiblingBelow (p, w))
+	    {
 		p = findValidStackSiblingBelow (sibling, p);
 
-	    /* if we found a valid sibling under the active window, it's
-	       our new sibling we want to stack above */
-	    if (p)
-		sibling = p;
+		/* if we found a valid sibling under the active window, it's
+		   our new sibling we want to stack above */
+		if (p)
+		  sibling = p;
+	    }
 	}
 
 	mask |= addWindowStackChanges (w, &xwc, sibling);
